capnslog/formatters.go

```diff
@@ -104,3 +127,31 @@ func (c *PrettyFormatter) Format(pkg string, l LogLevel, depth int, entries ...i
 func (c *PrettyFormatter) Flush() {
 	c.w.Flush()
 }
+
+type LogFormatter struct {
+	w      *bufio.Writer
+	logger *log.Logger
+	prefix string
+}
+
+func NewLogFormatter(w io.Writer, prefix string, flag int) *LogFormatter {
+	return &LogFormatter{
+		w:      bufio.NewWriter(w),
+		logger: log.New(w, "", flag), // don't use prefix here
+		prefix: prefix,               // save it instead
+	}
+}
+
+func (s *LogFormatter) Format(pkg string, l LogLevel, i int, entries ...interface{}) {
+	str := fmt.Sprint(entries...)
+	prefix := s.prefix
+	if pkg != "" {
+		prefix = fmt.Sprintf("%s%s: ", prefix, pkg)
+	}
+	s.logger.Output(5, fmt.Sprintf("%s%v", prefix, str)) // call depth is 5
+	s.Flush()
+}
+
+func (s *LogFormatter) Flush() {
+	s.w.Flush()
```

Comment:
The logger object (what the built in log provides) *does* write to w. Not 100% sure the flush is still needed, but I think it fits logically.